.vscode/launch.json

```diff
@@ -0,0 +1,43 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "GNSE_OPENOCD_STLINK_DBG",
+            "type": "cppdbg",
+            "request": "launch",
+            // Update `program` with your application elf path and make sure to update `setupCommands`
+            "program": "${workspaceRoot}/Software/build/debug/app/basic/basic.elf",
+            "args": [],
+            "stopAtEntry": true,
+            "cwd": "${workspaceRoot}",
+            "environment": [],
+            "externalConsole": false,
+            "debugServerArgs": "-f interface/stlink.cfg -f target/stm32wlx.cfg",
+            "serverLaunchTimeout": 20000,
+            "filterStderr": true,
+            "filterStdout": false,
+            "serverStarted": "Info\\ :\\ [\\w\\d\\.]*:\\ hardware",
+            "setupCommands": [
+                { "text": "cd ${workspaceRoot}/Software/build" },
+                { "text": "file debug/app/basic/basic.elf", "description": "load file", "ignoreFailures": false },
+                { "text": "target extended-remote localhost:3333", "description": "connect to target", "ignoreFailures": false },
+                { "text": "monitor reset halt", "description": "perform a reset and halt the target", "ignoreFailures": false },
+                { "text": "load", "description": "flash target", "ignoreFailures": false },
+                { "text": "monitor reset init", "description": "perform a reset and init the target", "ignoreFailures": false },
+                { "text": "set output-radix 16", "description": "set the default numeric base to 16", "ignoreFailures": false }
+            ],
+            "logging": {
+                "moduleLoad": true,
+                "trace": true,
+                "engineLogging": true,
+                "programOutput": true,
+                "exceptions": true
+            },
+            "MIMode": "gdb",
+            // Update `miDebuggerPath` with your compiler gdb path
+            "miDebuggerPath": "/Users/USERNAME/opt/gcc-arm-none-eabi-6-2017-q2-update/bin/arm-none-eabi-gdb",
```

Comment:
Can this be evaluated from `PATH` instead?

Reply:
I thought about that (like ${workspaceRoot}) but I think people install the toolchain in different locations, especially in windows environments.

Reply:
PATH also exists under Windows

Reply:
In any case, if this file is under source control, and you require people to edit the path here, there's a diff. That may lead to people accidentally committing this file in pull requests etc. So if we can avoid requiring editing this file, let's do that.

Reply:
Yes I agree with you that people accidentally can do this.
There is so many moving parts when it comes to configs and we can't force people to install their tools in a specific place and fix the bigger problem of operating system different file structure design and tools location.

I have been reading on this,
So we have two options:
1. Add the files and then ignore them in the .gitignore
2. Ask developer to use something like this command `git update-index --skip-worktree ./.vscode/settings.json`

I like the discussions in [here](https://stackoverflow.com/questions/13630849/git-difference-between-assume-unchanged-and-skip-worktree#) and [here](https://stackoverflow.com/questions/9794931/keep-file-in-a-git-repo-but-dont-track-changes) about this.

Let me know what you think?

Reply:
> There is so many moving parts when it comes to configs and we can't force people to install their tools in a specific place

Hence the suggestion to use `PATH`. Did you try that?

Reply:
No I didn't
I saw that the official documentation says you need to set it up manually, so I didn't bother 

https://code.visualstudio.com/docs/cpp/cpp-debug

I can try if you think it adds value., but do you think we should ask people to add their toolchain to path and go on to explain how to do that for each operating system?

Reply:
I tried the following:

```json
// Update `miDebuggerPath` with your compiler gdb path
"miDebuggerPath": "${PATH}/arm-none-eabi-gdb",
// Update `debugServerPath` with your openocd binary location
"debugServerPath": "openocd"
```
it says the value is invalid

![image](https://user-images.githubusercontent.com/24204263/140522462-2a286bc2-bb66-40c6-8e2e-b131b903016b.png)

I tried this an it works, but only because I made sure that `arm-none-eabi-gdb` and `openocd` binaries are at `/usr/local/bin`
```json
"miDebuggerPath": "arm-none-eabi-gdb",
"debugServerPath": "openocd"
````

Let me know if the is what you prefer

Reply:
This makes sense since `echo $PATH` gives me `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/opt/X11/bin`

Reply:
Yes, but this is really how all commands are being evaluated.

The only thing is that `arm-none-eabi-gdb` doesn't indicate the version.

What do similar repositories use as VS Code's `settings.json` and `launch.json` files?

Reply:
They tend not to add a launcher like

https://github.com/azure-rtos/getting-started/tree/master/.vscode

and the official documentation asks to add the paths manually

https://code.visualstudio.com/docs/cpp/cpp-debug

Reply:
OK, let's stick with that then.

The thing is, that if users have set this up already correctly in their user settings, our repository settings override that, and may stop things from working.

So be careful with repo settings; only things that are generic for all users, otherwise just document.

Reply:
(2) is a no-go really. Aren't there generic settings that we can commit in this repo? Then user settings do the rest

Reply:
Ok then, let's  leave it at using the documentation as mentioned in the first proposal.